service/index/reader.go

```diff
@@ -22,23 +22,24 @@ import (
 
 	"github.com/onflow/flow-go/ledger"
 	"github.com/onflow/flow-go/model/flow"
-
-	"github.com/optakt/flow-dps/service/storage"
 )
 
 // Reader implements the `index.Reader` interface on top of the DPS server's
 // Badger database index.
 type Reader struct {
+	storage StorageLibrary
+
 	db *badger.DB
 }
 
 // NewReader creates a new index reader, using the given database as the
 // underlying state repository. It is recommended to provide a read-only Badger
 // database.
-func NewReader(db *badger.DB) *Reader {
+func NewReader(db *badger.DB, lib StorageLibrary) *Reader {
```

Comment:
If you call it `lib` here, then the field should also be called `lib`. Let's keep it consistent. Would also suggest the same order as the fields on the reader.

Reply:
Then I guess i'd rather call it `storage`. For me `lib` is too generic, it could mean anything 🤔 WDYT?